tests/ttnn/distributed/test_distributed_reshape.cpp

```diff
@@ -0,0 +1,290 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+
+#include <cstddef>
+#include <ttnn/core.hpp>
+#include <ttnn/distributed/api.hpp>
+#include "tests/tt_metal/test_utils/env_vars.hpp"
+
+namespace ttnn::distributed::test {
+
+// Helper function to check test environment
+void check_test_environment() {
+    auto slow_dispatch = getenv("TT_METAL_SLOW_DISPATCH_MODE");
+    const auto arch = tt::get_arch_from_string(tt::test_utils::get_umd_arch_name());
+    const size_t num_devices = tt::tt_metal::GetNumAvailableDevices();
+    if (slow_dispatch) {
+        GTEST_SKIP() << "Skipping Multi-Device test suite, since it can only be run in Fast Dispatch Mode.";
+    }
+    if (num_devices < 8 or arch != tt::ARCH::WORMHOLE_B0) {
+        GTEST_SKIP() << "Skipping T3K Multi-Device test suite on non T3K machine.";
+    }
+}
+
+static std::vector<chip_id_t> get_physical_device_ids(const MeshDevice& mesh) {
```

Comment:
`static` keyword should be implied here?

Reply:
updated.